Compilation errors when including both WaveFolder.h and ResonantFilter.h

Hi,
The following sketch produces multiple compile errors

> #include <MozziGuts.h>
> #include <WaveFolder.h>
> #include <LowPassFilter.h>
> 
> void setup() {
>   // put your setup code here, to run once:
> }
> 
> void loop() {
>   // put your main code here, to run repeatedly:
> }
> 

The errors are repeatable on two PCs and take the following form :-

> _In file included from /home/john/Arduino/libraries/Mozzi/ResonantFilter.h:16:0,
>                  from /home/john/Arduino/libraries/Mozzi/LowPassFilter.h:16,
>                  from /home/john/Desktop/LPFandWF/LPFandWF.ino:3:
> /home/john/Arduino/libraries/Mozzi/IntegerType.h:2:32: error: redefinition of 'struct IntegerType<BYTES>'
>  template<uint8_t BYTES> struct IntegerType {_
>                                 ^~~~~~~~~~~

_

After some playing around I eliminated the compilation errors by adding #ifndef /#define directives to the file IntegerType.h. I must admit, I'm a bit out of my depth here so I thought I'd better log it for someone else to take a look at. 

Many thanks
John

--- a/IntegerType.h
+++ b/IntegerType.h
@@ -1,3 +1,5 @@
+#ifndef INTTYPE_H_
+#define INTTYPE_H_
 
 template<uint8_t BYTES> struct IntegerType {
     // at an odd value, such as 3 bytes? Add one more byte (up to at most 8 bytes)..
@@ -25,3 +27,5 @@ template<> struct IntegerType<8> {
     typedef uint64_t unsigned_type;
     typedef int64_t signed_type;
 };
+
+#endif /* INTTYPE_H_ */

